### PR TITLE
Loader fixes and improvements

### PR DESCRIPTION
**Pull request by Leopard20**
- [x] Use a better and safer alternative to `IsBadReadPtr`
- [x] Fix loader for vc143
- [x] Fix hashmap poolAlloc

### CODE REVIEW

**Comment by killerswin2** [comment]
Linux good?

**Comment by Leopard20** [comment]
> Linux good?

Fixed just now

**Comment by killerswin2** [comment]
lol answered my own question

**Comment by Leopard20** [comment]
Looks like Linux has been failing due to another issue too.

**Comment by Avokadomos** [comment]
This fixed the issues I had with HashMaps on Win64.

File: src/host/loader/ptraccess.h [hunk 1]
```diff
@@ -2,6 +2,46 @@
 #pragma once
 #if defined(_WIN32) || defined(_WIN64)
 #include <windows.h>
+#ifdef __cplusplus
+// From: https://stackoverflow.com/questions/496034/most-efficient-replacement-for-isbadreadptr
+// Check memory address access
+static const DWORD dwForbiddenArea = PAGE_GUARD | PAGE_NOACCESS;
```
Comment (dedmen) [inline src/host/loader/ptraccess.h:8]: Static const should maybe be upper case. Don't know if intercept has other examples of that?

**Review by dedmen — APPROVED** [review]

File: src/host/loader/ptraccess.h (hunk 1 above)
Comment (Leopard20) [reply src/host/loader/ptraccess.h:8]: Dunno. I literally copy pasted the code from that stack overflow link 😅 (and added the static)
I should probably replace it with constexpr tho?

**Pull request merged**